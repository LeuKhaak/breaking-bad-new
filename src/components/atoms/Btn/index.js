import React from "react";
import * as styles from "./styles.module.scss";
import cn from "classnames";

const Btn = ({
  location,
  disp,
  btnText,
  icon,
  btnStyle,
  type,
  color,
  onClick,
  arg,
  arg2,
  id,
}) => {
  const classBtn = cn(
    styles.btn,
    styles[btnStyle],
    styles[color],
    styles[type]
  );
  const onClick1 = () => {
    if (onClick) onClick(arg, arg2);
  };
  return (
    <button
      className={classBtn}
      id={id}
      style={
        location && location !== "/catalog"
          ? { background: "none" }
          : disp
          ? { display: disp }
          : {}
      }
      onClick={() => {
        onClick1();
      }}
    >
      {icon ? (
        <img
          className={styles.icon}
          src={icon}
          alt="#"
          style={{ fill: "red" }}
        />
      ) : (
        ""
      )}
      <span className={styles.linkText}>{btnText}</span>
    </button>
  );
};

export default Btn;
