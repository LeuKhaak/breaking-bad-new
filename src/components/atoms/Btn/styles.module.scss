@import "../../../assets/styles/styles.scss";

.btn {
  cursor: pointer;
  padding: 0.65rem 1.25rem;
  box-sizing: border-box;
  border-radius: 0.25rem;
  display: flex;
  align-items: center;
  border: none;
  background-color: transparent;
  .icon {
    margin-right: 0.75rem;
    width: 1rem;
  }
}
.toCatalog {
  background: $gold;
  border: 0.09rem solid $gold;
}

.toggle {
  padding: 0;
  .icon {
    cursor: pointer;
    margin-right: 0;
    width: 1.5rem;
  }
}
.pageNumber {
  @include pageButton;
  padding: 0;
}

.pageNumberChecked {
  @include pageButton;
  padding: 0;
  color: $gold;
  background: $white;
  box-shadow: 0 0.25rem 0.5rem $shadowDark;
  border-radius: 0.25rem;
}

.buttonNext {
  @include pageButton;
  transform: rotateY(180deg);
}
