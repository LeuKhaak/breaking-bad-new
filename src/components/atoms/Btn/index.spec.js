import React from "react";
import { shallow } from "enzyme";
import Btn from ".";

let component = null;
const onClick = jest.fn();
const setComponent = (props) => shallow(<Btn {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("Btn testing:", () => {
  test("is Btn contain icon", () => {
    component = setComponent({
      icon: "./test.jpg",
      btnStyle: "testStyle",
    });
    expect(component).toMatchSnapshot();
  });

  test("is Btn contain text", () => {
    component = setComponent({
      btnText: "testText",
      btnStyle: "testStyle",
    });
    expect(component).toMatchSnapshot();
  });

  test("is Btn contain text and icon", () => {
    component = setComponent({
      icon: "./test.jpg",
      btnText: "testText",
      btnStyle: "testStyle",
    });
    expect(component).toMatchSnapshot();
  });

  test("is Btn contain nothing", () => {
    expect(component).toMatchSnapshot();
  });

  test("is Btn not displayed", () => {
    component = setComponent({
      btnText: "testText",
      disp: "none",
    });
    expect(component).toMatchSnapshot();
  });

  test("is Btn has backgroung", () => {
    component = setComponent({
      btnText: "testText",
      location: "/catalog",
    });
    expect(component).toMatchSnapshot();
  });

  test("is Btn hasn`t backgroung", () => {
    component = setComponent({
      icon: "./test.jpg",
      location: "/person/1",
    });
    expect(component).toMatchSnapshot();
  });

  test("onClickFunc", () => {
    component = setComponent({
      onClick: onClick,
      icon: "./test.jpg",
    });
    component.find("button").simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
  test("is Btn has className", () => {
    component = setComponent({
      btnText: "testText",
      btnStyle: "buttonNext",
    });
    expect(component.find("buttonNext")).toBeTruthy();
  });
});
