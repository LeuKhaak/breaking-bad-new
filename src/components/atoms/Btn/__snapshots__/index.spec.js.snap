// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn testing: is Btn contain icon 1`] = `
<button
  className=""
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="#"
    src="./test.jpg"
    style={
      Object {
        "fill": "red",
      }
    }
  />
  <span />
</button>
`;

exports[`Btn testing: is Btn contain nothing 1`] = `
<button
  className=""
  onClick={[Function]}
  style={Object {}}
>
  <span />
</button>
`;

exports[`Btn testing: is Btn contain text 1`] = `
<button
  className=""
  onClick={[Function]}
  style={Object {}}
>
  <span>
    testText
  </span>
</button>
`;

exports[`Btn testing: is Btn contain text and icon 1`] = `
<button
  className=""
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="#"
    src="./test.jpg"
    style={
      Object {
        "fill": "red",
      }
    }
  />
  <span>
    testText
  </span>
</button>
`;

exports[`Btn testing: is Btn has backgroung 1`] = `
<button
  className=""
  onClick={[Function]}
  style={Object {}}
>
  <span>
    testText
  </span>
</button>
`;

exports[`Btn testing: is Btn hasn\`t backgroung 1`] = `
<button
  className=""
  onClick={[Function]}
  style={
    Object {
      "background": "none",
    }
  }
>
  <img
    alt="#"
    src="./test.jpg"
    style={
      Object {
        "fill": "red",
      }
    }
  />
  <span />
</button>
`;

exports[`Btn testing: is Btn not displayed 1`] = `
<button
  className=""
  onClick={[Function]}
  style={
    Object {
      "display": "none",
    }
  }
>
  <span>
    testText
  </span>
</button>
`;
